homeassistant/components/harmony/data.py

```diff
@@ -159,10 +163,12 @@ async def async_start_activity(self, activity: str):
             )
             return
 
+        await self.async_lock_start_activity()
         try:
             await self._client.start_activity(activity_id)
         except aioexc.TimeOut:
             _LOGGER.error("%s: Starting activity %s timed-out", self.name, activity)
+            self.async_unlock_start_activity()
```

Comment:
Should this be in a finally clause?

Reply:
We don't the lock to release until the callback happens.

Reply:
Hmm. If we get an unhandled exception we should release it though

Reply:
Just dug into it and I don't think any other exceptions should happen unless something is really wrong and we disconnect, but we will clear the lock at that point anyways.  So probably ok as-is.